pysal/contrib/viz/plot.py

```diff
@@ -0,0 +1,47 @@
+"""
+Moran Plot using PySAL and Matplotlib
+
+"""
+
+__author__ = "Marynia Kolak <marynia.kolak@gmail.com>"
+
+import pandas as pd
+import numpy as np
+import pysal as ps
+import matplotlib.pyplot as plt
+
+
+def Moran(var, w, **kwargs):
```

Comment:
I think that we may have to think about some of the functionality of the
moran's I input in the moran plot function. For example, the issue of
standardization (or zx and zy in the analytic moran function).. Unless the
plot function is a function within the class of moran (which wouldn't make
sense for our purposes), I don't see how else we could pull values zx and
zy from the moran function, as it's not defined outside of that class. In
testing I can create a plot with just the MI value, as I've assumed I've
previously declared my variable and weight -- I can use the MI as the
slope, but I am still required to re-calculate the plotted values.

In moran plots for spdep in R
<http://www.inside-r.org/packages/cran/spdep/docs/moran.plot>and in GeoDa,
you enter the variable and weight to initiate. I know the goal here is to
try out different moran options, and then plot when we're satisfied. So for
our purposes, should we stick with a leaner "MI" input but still redo some
of the calculations as needed? Or do we add more inputs, as now? How
"cooked" do we want this -- soft or hard boiled?

Wanted to really hammer this piece out, as it will serve as a useful
guideline for next views...



On Sun, Apr 24, 2016 at 3:31 AM, Dani Arribas-Bel <notifications@github.com>
wrote:

> In pysal/contrib/viz/plot.py
> <https://github.com/pysal/pysal/pull/797#discussion_r60842435>:
>
> > @@ -0,0 +1,47 @@
> > +"""
> > +Moran Plot using PySAL and Matplotlib
> > +
> > +"""
> > +
> > +__author__ = "Marynia Kolak <marynia.kolak@gmail.com>"
> > +
> > +import pandas as pd
> > +import numpy as np
> > +import pysal as ps
> > +import matplotlib.pyplot as plt
> > +
> > +
> > +def Moran(var, w, **kwargs):
>
> Two questions:
>
>    1. What'd be **kwargs? I don't see inside the method where those could
>    be passed.
>    2. Rather than asking for the var and the w object, I think it'd be
>    nicer to have it only take a Moran object. Then we can pull in all the
>    parts we need for the plot. The rationale for that is that the user would
>    compute a Moran object, check the value and, if (s)he wanted to get a
>    visual representation would just pass that object to this method. This
>    approach would also allow us to easily hook up this method inside Moran
>    so you could (down the line) do things like:
>
> m = ps.Moran(y, w)
> m.plot()
>
> And get the plot.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/pysal/pysal/pull/797/files/2fb3d3b7eaa9d6bb2032f7393d54f04478937eb6#r60842435>
>